### DISCUSSION TITLE
how to disable resizing of panels.

### DISCUSSION THREAD

**Discussion by husayt** [mathuo/dockview#457] [Q&A] [answered]
Hi @mathuo, 
thank you again for an amazing work here.
I have the following user requirement and was wondering if there is an easy way to achieve it.

Sometimes user wants to lock the layout, to disable resizing of all panels. Is it currently possible?

Thank you

**Comment by husayt** [comment]
@mathuo what essentially is needed is to disable dnd. Is there a way to do it currently? or is this a new feature request?

this is really essential for building dashboards

Thanks

**Reply by techbech** [reply]
Looking at the documentation it's not possible right now.

An immediate issue with locking would be if the user resizes the window or the dockview element gets resized in some other way. If the layout is locked, the user can't resize it. Would that be something you expect Dockview to be able to handle? Maybe by resizing every panel proportionally down until their minimum sizes.

**Reply by husayt** [reply]
you see current behavior is all fine, the only thing needed is to disallow resizing of panels by user via mouse. Grid control's resizing of panels on container's resize is just fine.

**Comment by mathuo** [comment, marked as answer]
I guess there are two features here:
1. Disable mouse-interaction based resizing
    - I think this could be achieved entirely through CSS making it a fairly small change that is required.
2. Disable the dnd drop functionality 
    - This one is more involved but actually seems to be fairly similar to an existing issue [here](https://github.com/mathuo/dockview/issues/395) which I have been doing some work on which is to expose a function allowing control of whether dnd overlays are shown, in your case this function would just disable all overlays.

**Reply by husayt** [reply]
yes, both seem are good to have and would solve my problem. Second option is the more powerful though. I think they are both valid solutions and not mutually exclusive.

**Reply by husayt** [reply]
> this

thanks. first one, worked very well for my case. just one line css change.

**Reply by mathuo** [reply]
Created https://github.com/mathuo/dockview/issues/460 to incorperate a locked flag, other asks will be included in https://github.com/mathuo/dockview/issues/395.

**Reply by StringKe** [reply]
How to lock a single panel to do a sidebar-like effect?
Do not allow panels to move, close.

**Reply by Code-Victor** [reply]
@StringKe where you able to figure it out

**Comment by sarabs3js** [comment]
Disable mouse-interaction based resizing

What is that one line change of css, please mention I have same requirement

**Comment by SciManJD** [comment]
I also need some help with this issue. Please, write this magic css line...
